crates/ln-dlc-node/src/fee_rate_estimates.rs

```diff
@@ -0,0 +1,100 @@
+use anyhow::Result;
+use autometrics::autometrics;
+use bdk::FeeRate;
+use lightning::chain::chaininterface::ConfirmationTarget;
+use lightning::chain::chaininterface::FeeEstimator;
+use lightning::chain::chaininterface::FEERATE_FLOOR_SATS_PER_KW;
+use std::collections::HashMap;
+use tokio::sync::RwLock;
+
+pub struct FeeRateEstimates {
+    client: esplora_client::AsyncClient,
+    fee_rate_cache: RwLock<HashMap<ConfirmationTarget, FeeRate>>,
+    fallbacks: RwLock<FeeRateFallbacks>,
+    runtime_handle: tokio::runtime::Handle,
+}
+
+#[derive(Debug)]
+pub struct FeeRateFallbacks {
+    pub normal_priority: u32,
+    pub high_priority: u32,
+}
+
+impl Default for FeeRateFallbacks {
+    fn default() -> Self {
+        Self {
+            normal_priority: 2000,
+            high_priority: 5000,
+        }
+    }
+}
+
+impl FeeRateEstimates {
+    pub fn new(esplora_url: String, runtime_handle: tokio::runtime::Handle) -> Self {
+        let client = esplora_client::AsyncClient::from_client(esplora_url, reqwest::Client::new());
+
+        let fee_rate_cache = RwLock::new(HashMap::default());
+        let fallbacks = RwLock::new(FeeRateFallbacks::default());
+
+        Self {
+            client,
+            fee_rate_cache,
+            fallbacks,
+            runtime_handle,
+        }
+    }
+
+    pub(crate) fn get(&self, target: ConfirmationTarget) -> FeeRate {
+        let (cache, fallbacks) = self.runtime_handle.block_on(async {
+            (
+                self.fee_rate_cache.read().await,
+                self.fallbacks.read().await,
+            )
+        });
```

Comment:
🤔 this seems odd. we can probably get rid of the `block_on` call here? 

(it was the earlier, but I'd like to flag this).

Reply:
The way forward here is to move away from `tokio::sync::RwLock` in favour of `std::sync::RwLock`. Thanks for leading me there.